src/sparse_ir/gauss.py

```diff
@@ -33,17 +33,12 @@ def __init__(self, x, w, x_forward=None, x_backward=None, a=-1, b=1):
 
     def reseat(self, a, b):
         """Reseat current quadrature rule to new domain"""
-        if self.a != -1 or self.b != 1:
-            raise NotImplementedError("only supporting unscaled for now")
-
-        extent = 0.5 * (b - a)
-        midpoint = 0.5 * (b + a)
-        x_new = extent * self.x
-        x_new += midpoint
-        x_forward = extent * self.x_forward
-        x_backward = extent * self.x_backward
-        w_new = extent * self.w
-        return Rule(x_new, w_new, x_forward, x_backward, a, b)
+        scaling = (b - a) / (self.b - self.a)
+        x = (self.x - self.a) * scaling + a
```

Comment:
Careful: that potentially leads to bad cancellation for x around 0!

Fix - this is untested though:
```suggestion
        x = scaling * (self.x - (self.b + self.a)/2) + (b + a)/2
```

Reply:
I consulted Mathematica for the algebra and it seems like `shift` should be `a - self.a * scaling == (self.a * b - self.b * a) / (self.a - self.b)` instead. Would that work from an cancellation avoidance-perspective?